releases/ReleaseCompatibility.md

```diff
@@ -0,0 +1,103 @@
+However, we do *not* guarantee compatibility between `n` and `n+1` in this scheme (although we would seek to minimise breakage).
+If we have to do `v.n.m.post[i]` releases, we will only support the final `post` release in the chain.
+
+In order to support less mature functionality, we should also add
+a `fairlearn.experimental` package (see [a similar namespace in
+SciKit-Learn](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.experimental)).
+Anything in there will be subject to breaking at any time.
+Using an `experimental` namespace would not have helped with our
+current set of breaks, since the changes were being made to core
+functionality.
+Rather, this is to give future developers a space where they can
+get feedback on new functionality without immediately being
+committed to supporting their speculative design decisions.
+
+At the time of writing, it appears that the dashboard code does
+not use namespaces.
+However, namespaces are supported in TypeScript, and as we develop
+the UX code, we should introduce a similar distinction.
+
+To monitor the required backwards compatibility, each new release
+will have a build pipeline associated with it.
+This pipeline will run the *tests associated with the release tag*
+against the version of Fairlearn in `master`.
+In keeping with the note about `post[i]` releases above, the build will only run against the last release in each release branch.
+The build will become part of the PR Gate for `master`, except when
+we are moving from `v0.n` to `v0.n+1`.
+
+## Revised Branching and Release Policy
+
+Our recent releases have been made from `master`, without
+making a release branch.
+While this approach has desirable properties, using release branches
+will work better with the more robust support policy described above.
+
+For all the `master` and `release` branches, we will require a linear
+history in GitHub.
+This forbids plain `merge` commits, and we will prefer squash merges
+to rebases.
+
+We will keep the `master` branch at `v0.m.n.dev0` at all times,
+indicating that `master` is under active development (although we will
+always seek to keep `master` in a shippable condition).
+Releases will occur from branches.
+
+To create a new release:
+1. Create a branch `release/v.0.n.m`
+1. Remove the `dev0` suffix from the version on the release branch
+   - Bump `master` to `v0.n.m+1.dev0`
+1. Run the release pipeline on the new release branch
+1. Create a GitHub Release corresponding to the new package
+   (hopefully this can be automated)
+1. Create the new build pipeline to monitor backwards compatibility
+
+### Post Releases
+
+We should only put out `post[i]` releases for essential fixes (either
+to algorithms or code) - no new features are allowed.
+In general, fixes should be made in `master` and individually moved
+to the appropriate release branch as required.
+The move will probably best be done with `git rebase -i`, in order to
```

Comment:
Did you mean git cherry-pick?

Reply:
No - scikitlearn uses `git rebase -i` instead. I know that cherry-picking gave trouble before. I'm not sure if that was just my general ineptitude, or whether something more fundamental about it.

Reply:
the `rebase` done in sklearn is a _reverse rebase_. A bit confusing, I know :D but it basically does the `git cherry-pick` :P

Reply:
When it comes to `git` phrases like "basically does the same" cause me a certain amount of trepidation....

It is possible that the issues I had with the early release were due to me working on a clone of Fairlearn and not a fork. For now, I've tried to explain a bit more, and put in a link to the `scikit-learn` doc. I feel that this might be something which is somewhat obscure until we actually try to do it.

Reply:
... and when we try it it would be great to document the explicit steps :-)

Thanks for your clarifications!

Reply:
I've added a note about doing the documentation once we're sure of the steps.